src/plugins/ui_feature/selection_inputs_manager.ts

```diff
@@ -98,9 +98,9 @@ export class SelectionInputsManagerPlugin extends UIPlugin {
         break;
       case "FOCUS_RANGE":
       case "CHANGE_RANGE":
-        if (cmd.id !== this.focusedInputId) {
-          const input = this.inputs[cmd.id];
-          const range = input.ranges.find((range) => range.id === cmd.rangeId);
+        const input = this.inputs[cmd.id];
```

Comment:
it seems weird to me  -  If we are already focused on the range, then dispatching a "focus_range" should have 0 effect (same goes for change range) since by definition, we already captured the selection.  Isn't there something else that we are missing? 
It looks like we currently can reach a state where we have `this.focusedInputId` defined but without having the control of the selection ,which would be the root problem and we are only masking it because we know that the components dispatch some specific commands. Let me know what you think about this :)

Reply:
If I understand correctly, `FOCUS_RANGE` makes sense even when clicking on different ranges in the same SelectionInput component. This is because we need to capture a zone that allows us to select the next cell whenever the user presses the arrow keys. right?

Reply:
Yes, you are right - I (again) mistook the generic selection input with one of the range inputs.